_src/static/sass/components/_headings.scss

```diff
@@ -0,0 +1,128 @@
+// ------------------------------
+// edX Pattern Library: Components - Headings
+
+// About: Contains base styling for headings, relying on Bourbon's mathematical scales for precision.
+
+// Scale:
+// 12, 14, 16, 18, 21, 24, 32, 43
+// ----------------------------
+
+// #GLOBALS
+// #HEADINGS
+// H1
+// H2
+// H3
+// H4
+// H5
+// H6
+// ----------------------------
+
+
+// ----------------------------
+// #GLOBALS
+// ----------------------------
+.hd-1,
+.hd-2,
+.hd-3,
+.hd-4,
+.hd-5,
+.hd-6 {
+    font-family: $headings-base-font-family;
+    font-weight: $headings-base-font-weight;
+    color: $headings-base-color;
+
+    &.emphasized,
+    &.primary {
+        font-weight: $headings-primary-font-weight;
+        color: $headings-primary-color;
+    }
+
+    &.deemphasized,
+    &.secondary {
+        font-weight: $headings-secondary-font-weight;
+        color: $headings-secondary-color;
+    }
+}
+
+.hd-5,
+.hd-6 {
+    font-weight: $headings-primary-font-weight;
+    text-transform: uppercase;
+}
+
+
+// ----------------------------
+// #HEADINGS
+// ----------------------------
+// H1
+// ----------------------------
+.hd-1 {
+    font-size: font-size(xx-large);
+    line-height: line-height(xx-large);
+}
+
+
+// ----------------------------
+// H2
+// ----------------------------
+.hd-2 {
+    font-size: font-size(x-large);
+    line-height: line-height(x-large);
+}
+
+
+// ----------------------------
+// H3
+// ----------------------------
+.hd-3 {
+    font-size: font-size(large);
+    line-height: line-height(large);
+
+    &.deemphasized,
+    &.secondary {
+        color: palette(grayscale, base);
+    }
+}
+
+
+// ----------------------------
+// H4
+// ----------------------------
+.hd-4 {
+    font-weight: font-weight(bold);
+    font-size: font-size(mid-large);
+    line-height: line-height(mid-large);
+
+    &.deemphasized,
+    &.secondary {
+        color: palette(grayscale, mid-dark);
+    }
+}
+
+
+// ----------------------------
+// H5
+// ----------------------------
+.hd-5 {
+    font-size: font-size(mid-small);
+    line-height: line-height(mid-small);
+
+    &.deemphasized,
+    &.secondary {
+        color: palette(grayscale, mid-dark);
+    }
+}
+
+
+// ----------------------------
+// H6
+// ----------------------------
+.hd-6 {
+    font-size: font-size(base);
+    line-height: line-height(base);
+
+    &.deemphasized,
+    &.secondary {
+        color: palette(grayscale, mid-dark);
+    }
+}
```

Comment:
Good question. I tend to like the simplicity of a two level scale for headings:

* base - default headings (including more bolder weights/colors for the more higher level headings)
* de-emphasized - for the more subdued/downplayed versions

Primary and secondary seem create a more flexible but not as clear (especially with our class names for headings also adding a # scale, i.e. .hd-1-6) scale. I'd recommend losing the primary and secondary classes.